src/Providers/BlockServiceProvider.php

```diff
@@ -171,4 +177,23 @@ public function registerMeta(): void
             }, $this->app[$block]->blockMeta());
         }
     }
+
+    /**
+     * Retrieve the block icon.
+     *
+     * @param $icon
+     *
+     * @return string
+     */
+    public function getIcon($icon): string
+    {
+        if ( str_contains( $icon, '<svg' ) ) {
+            $xml = simplexml_load_string($icon);
+            $svg_obj = ['svg' => $xml];
```

Comment:
let's add a little check

```
if (false === $xml) {
  return $icon;
}
```

Reply:
Sure, I also cleaned it out a bit. Change pushed.